tests/integration/test_backup_restore_s3/test.py

```diff
@@ -452,3 +452,57 @@ def test_backup_to_zip():
     backup_name = new_backup_name()
     backup_destination = f"S3('http://minio1:9001/root/data/backups/{backup_name}.zip', 'minio', 'minio123')"
     check_backup_and_restore(storage_policy, backup_destination)
+
+
+def test_user_specific_auth(start_cluster):
+    def create_user(user):
+        node.query(f"CREATE USER {user}")
+        node.query(f"GRANT CURRENT GRANTS ON *.* TO {user}")
+
+    create_user("superuser1")
+    create_user("superuser2")
+    create_user("regularuser")
+
+    node.query("CREATE TABLE specific_auth (col UInt64) ENGINE=Memory")
+
+    assert "Access Denied" in node.query_and_get_error(
+        "BACKUP TABLE specific_auth TO S3('http://minio1:9001/root/data/backups/limited/backup1.zip')"
+    )
```

Comment:
This test seems incorrect, see failure
https://s3.amazonaws.com/clickhouse-test-reports/0/34e8e673e9241a31d9f0063f81f09a924ff0de00/integration_tests__tsan_.html

Reply:
It seems it pulled somehow credentials from `EC2MetadataService`. I will check it out, thanks for notifying.